players.js

```diff
@@ -235,42 +235,54 @@ class Player {
 		}
 		return ""
 	}
-	getArtUrlIcon(size){
+	getArtUrlIcon(size, iconPlace, padding){
+		const shellVersion = Number.parseInt(imports.misc.config.PACKAGE_VERSION.split('.'));
 		const url = this.stringFromMetadata("mpris:artUrl",this.metadata);
+
+		let icon_left_padding = 0;
+		let icon_right_padding = 0;
+		if (shellVersion >= 3)
+			if (iconPlace == "right")
+				icon_left_padding = padding
+			else if (iconPlace == "left")
+				icon_right_padding = padding
+
 		if(url.length>0)
 			this.albumArt = new St.Icon({
 				gicon: Gio.Icon.new_for_string(url),
 				style_class: 'system-status-icon',
 				icon_size: size,
-				style: "padding: 0px;"
+				style: "padding: 0px; padding-left: " + icon_left_padding + "px;padding-right: " + icon_right_padding + "px;"
 			})
 		else this.albumArt = null;
 
 		return this.albumArt
 	}
-	getIcon(desktopApp){
-		const settings = ExtensionUtils.getSettings('org.gnome.shell.extensions.mpris-label');
-		const ICON_PLACE = settings.get_string('show-icon');
-		const Config = imports.misc.config;
+	getIcon(iconPlace, symbolicIcon, padding){
+		const shellVersion = Number.parseInt(imports.misc.config.PACKAGE_VERSION.split('.'));
 
 		let icon_left_padding = 0;
 		let icon_right_padding = 0;
-		if (Config.PACKAGE_VERSION.startsWith("3."))
-			if (ICON_PLACE == "right")
-				icon_left_padding = 3
-			else if (ICON_PLACE == "left")
-				icon_right_padding = 3
+		if (shellVersion >= 3)
```

Comment:
Well now that i think of it. This statement will always be true.

There's no point of this workaround and we can safely remove it.